pandas/core/missing.py

```diff
@@ -964,3 +974,107 @@ def _rolling_window(a: np.ndarray, window: int):
     shape = a.shape[:-1] + (a.shape[-1] - window + 1, window)
     strides = a.strides + (a.strides[-1],)
     return np.lib.stride_tricks.as_strided(a, shape=shape, strides=strides)
+
+
+def _can_hold_element(values, element: Any) -> bool:
+    """
+    Expanded version of core.dtypes.cast.can_hold_element
+    """
+    from pandas.core.arrays import (
+        ExtensionArray,
+        FloatingArray,
+        IntegerArray,
+    )
+
+    if isinstance(values, ExtensionArray):
+        if hasattr(values, "_validate_setitem_value"):
+            # NDArrayBackedExtensionArray
+            try:
+                # error: "Callable[..., Any]" has no attribute "_validate_setitem_value"
+                values._validate_setitem_value(element)  # type: ignore[attr-defined]
+                return True
+            except (ValueError, TypeError):
+                return False
+        else:
+            # other ExtensionArrays
+            return True
+    else:
+        element = extract_array(element, extract_numpy=True)
+        if isinstance(element, (IntegerArray, FloatingArray)):
+            if element._mask.any():
+                return False
```

Comment:
not even for object dtype?

Reply:
(no longer relevant since now using `can_hold_element`)